### ISSUE TITLE
ref instead of id

### ISSUE THREAD

**Issue by zoransa** [TheDynomike/vue-script-component#1]
Hi you can use ref instead of id snd you do not need to pack template in two divs

**Comment by zoransa** [comment]
Another suggestion is just avoid using 'ad' because adBlocker might block it heuristically like they block iAB formats etc. Good component by the way.

**Comment by TheDynomike** [comment]
I agree removing the nested divs is probably a good idea and removing "adId" in source code is probably a good idea since the resulting component can be used for things outside of ad placement.

But, I'm not convinced with replacing id with ref is of value here since the id is not static.

I'll make these changes, thanks for the feedback!

**Comment by zoransa** [comment]
To be honest I made own component because I have some specific requirements but I wanted to day this is absolutely awesome idea... I saw script2 floating around people were doing some stuff with it but I never was happy with that solution.

**Comment by zoransa** [comment]
`<template>
    <div ref="element">
    </div>
</template>

<script>
    import postscribe from "postscribe";

    var ready = require('document-ready');
    export default {
        name: "AsyncJavascript",
        props: ["script64"],
        computed: {
            script() {
                return atob(this.script64);
            }
        },
        mounted() {
            let vm = this;
            ready(function () {
                let addEl = new Promise((resolve, reject) => {
                    postscribe(
                        vm.$refs.element,
                        `${vm.script}`,
                        {
                            done: function (x) {
                            }
                        }
                    );
                    resolve()
                });
                addEl.then(function (result) {
                    //do nothing
                })
            })
        }
    }
</script>

<style scoped>

</style>`

just <asyncjavascript script64="ZnVjayBpdCBkdWRlIQ=="></script>
